nncf/torch/nncf_network.py

```diff
@@ -422,12 +440,23 @@ def insert_at_point(self, point: PTInsertionPoint, fn_list: List[Callable]):
             assert norm_target_scope in norm_nncf_scopes  # Required for proper Recurrent/VariableRecurrent addressing
             if point.insertion_type == PTInsertionType.NNCF_MODULE_PRE_OP:
                 for fn in fn_list:
-                    nncf_module.register_pre_forward_operation(fn)
+                    handles.append(nncf_module.register_pre_forward_operation(fn))
             elif point.insertion_type == PTInsertionType.NNCF_MODULE_POST_OP:
                 for fn in fn_list:
-                    nncf_module.register_post_forward_operation(fn)
+                    handles.append(nncf_module.register_post_forward_operation(fn))
         else:
             raise RuntimeError("Unsupported insertion type: {}".format(point.insertion_type))
+        self._groups_vs_hooks_handlers[group].extend(handles)
+
+    def remove_hooks_group(self, group: HookGroups) -> None:
```

Comment:
Can I remove only one hook?

Reply:
Yes, in case you insert it manually by `NNCFNetwork` api

Reply:
Please describe a scenario when this one hook removal is required

Reply:
I think, nncf.strip is such scenario and any scenarios with collection custom statistics from the model.